src/main/java/org/cactoos/io/LoggingInputStream.java

```diff
@@ -121,8 +121,13 @@ public LoggingInputStream(
     @Override
     public int read() throws IOException {
         final byte[] buf = new byte[1];
-        this.read(buf);
-        return Byte.toUnsignedInt(buf[0]);
+        final int size;
+        if (this.read(buf) == -1) {
+            size = -1;
+        } else {
+            size = Byte.toUnsignedInt(buf[0]);
```

Comment:
@ilyakharlamov Let's use something like `value & 0xff` to get the byte. WDYT?

Reply:
@fabriciofx If you check the signature of java.lang.Byte.toUnsignedInt its contents is the following ```    public static int toUnsignedInt(byte x) {
        return ((int) x) & 0xff;
    }``` :)

Reply:
@ilyakharlamov 😄 right! So, let's just wait for #1023.